src/include/stir/VectorWithOffset.inl

```diff
@@ -177,15 +177,23 @@ typename VectorWithOffset<T>::iterator
 VectorWithOffset<T>::end() 
 {
   this->check_state();
-  return typename VectorWithOffset<T>::iterator(num+this->get_max_index()+1); 
+  // for empty vectors, the begin and end iterators need to point to the same place
+  if (length == 0)
+    return typename VectorWithOffset<T>::iterator(num+this->get_min_index());
```

Comment:
good point, will do!

Reply:
Hmm, good point... I got an warning from UBSAN here, and this change resolved it. I did debug through it and it made sense at the time. But now I'm no longer sure.

Reply:
ok. there could be a corner case here, as an empty `VectorWithOffset` is a weird thing. I'd have to check what `num` is set to etc. If you get the warnings, let's make the change (but if you don't, I wouldn't, as it could marginally slow things down to have the extra test).